stdlib/build.py

```diff
@@ -1,8 +1,80 @@
 #!/usr/bin/env python3.6
 # -*- coding: utf-8 -*-
+"""Types and functions to create and manipulate a :py:class:`.Build`, an instance of a
+:py:class:`.BuildManifest` for a specific version.
+"""
 
-import stdlib
+import os
+import shutil
+from typing import Dict
+
+_current_build = None
 
 
 class Build():
-    pass
+    """An instance of a :py:class:`.BuildManifest` for a specific version.
+
+    :param manifest: The :py:class:`.BuildManifest` this :py:class:`.Build` belongs to.
+    :type manifest: :py:class:`~.BuildManifest`
+
+    :param args: The arguments of this build. This is a subset of ``BuildManifest.versionized_args``,
+        the one corresponding to the version of this build.
+    """
+    def __init__(
+        self,
+        manifest,
+        args: Dict[str, str],
+    ):
+        self.manifest = manifest
+        self.args = args
+
+        self.semver = self.args['semver']
+        (self.major, self.minor, self.patch) = self.args['semver'].split('.')
+
+        from core.cache import get_download_cache, get_build_cache, get_install_cache
+        self.download_cache = get_download_cache(self)
+        self.build_cache = get_build_cache(self)
+        self.install_cache = get_install_cache(self)
+
+    def __str__(self):
+        return f'''{self.manifest.metadata.name} ({self.semver})'''
+
+    def build(self):
+        """Set the current build to ``self`` and execute the instructions contained in the :py:class:`.BuildManifest`.
+
+        :returns: An iterator over a list of :py:class:`.Package` s.
+            It forwards the return value of the instructions contained in the :py:class:`.BuildManifest`.
+        :returntype: ``List`` [ :py:class:`.Package` ]
```

Comment:
Well, is it a list, or an iterator then ?

Reply:
My bad, I didn't knew `typing` provide an `Iterator` class. I defaulted to `List` instead.